<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>455</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Snake_Robot_Controller</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>:/images/gui_png/rodu.png</normaloff>:/images/gui_png/rodu.png</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">background: #333;</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="minimumSize">
    <size>
     <width>800</width>
     <height>0</height>
    </size>
   </property>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>340</y>
      <width>241</width>
      <height>91</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="button_left">
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>80</height>
        </size>
       </property>
       <property name="mouseTracking">
        <bool>false</bool>
       </property>
       <property name="tabletTracking">
        <bool>false</bool>
       </property>
       <property name="acceptDrops">
        <bool>false</bool>
       </property>
       <property name="statusTip">
        <string/>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">background-repeat: no-repeat;</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/images/gui_png/special2.png</normaloff>:/images/gui_png/special2.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>80</width>
         <height>80</height>
        </size>
       </property>
       <property name="flat">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Fixed</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>34</width>
         <height>34</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="button_right">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>80</height>
        </size>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">width: 73px;
height: 68px;
flex-shrink: 0;
</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/images/gui_png/special1.png</normaloff>:/images/gui_png/special1.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>80</width>
         <height>80</height>
        </size>
       </property>
       <property name="autoDefault">
        <bool>false</bool>
       </property>
       <property name="default">
        <bool>false</bool>
       </property>
       <property name="flat">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="formLayoutWidget">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>66</y>
      <width>384</width>
      <height>412</height>
     </rect>
    </property>
    <layout class="QFormLayout" name="formLayout">
     <property name="verticalSpacing">
      <number>4</number>
     </property>
     <item row="0" column="0">
      <widget class="QPushButton" name="function_1">
       <property name="font">
        <font>
         <pointsize>20</pointsize>
         <weight>50</weight>
         <italic>false</italic>
         <bold>false</bold>
         <underline>false</underline>
         <strikeout>false</strikeout>
         <kerning>false</kerning>
        </font>
       </property>
       <property name="accessibleName">
        <string notr="true"/>
       </property>
       <property name="accessibleDescription">
        <string/>
       </property>
       <property name="layoutDirection">
        <enum>Qt::LeftToRight</enum>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="inputMethodHints">
        <set>Qt::ImhNone</set>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/images/gui_png/number1.png</normaloff>:/images/gui_png/number1.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>80</width>
         <height>80</height>
        </size>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
       <property name="autoRepeat">
        <bool>false</bool>
       </property>
       <property name="autoExclusive">
        <bool>false</bool>
       </property>
       <property name="autoDefault">
        <bool>false</bool>
       </property>
       <property name="default">
        <bool>false</bool>
       </property>
       <property name="flat">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="label1">
       <property name="font">
        <font>
         <family>Noto Sans</family>
         <pointsize>20</pointsize>
         <weight>75</weight>
         <italic>false</italic>
         <bold>true</bold>
         <strikeout>false</strikeout>
         <kerning>true</kerning>
        </font>
       </property>
       <property name="mouseTracking">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">
border-bottom: 1px solid orange;
color : white;
font-family: Noto Sans;
font-style: normal;
line-height: normal;</string>
       </property>
       <property name="text">
        <string>  Curve : 0 , Vel : 0</string>
       </property>
       <property name="textFormat">
        <enum>Qt::AutoText</enum>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="function_2">
       <property name="font">
        <font>
         <pointsize>20</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/images/gui_png/number2.png</normaloff>:/images/gui_png/number2.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>80</width>
         <height>80</height>
        </size>
       </property>
       <property name="flat">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="label2">
       <property name="font">
        <font>
         <family>Noto Sans</family>
         <pointsize>20</pointsize>
         <weight>75</weight>
         <italic>false</italic>
         <bold>true</bold>
         <strikeout>false</strikeout>
         <kerning>true</kerning>
        </font>
       </property>
       <property name="mouseTracking">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">
border-bottom: 1px solid orange;
color : white;
font-family: Noto Sans;
font-style: normal;
line-height: normal;</string>
       </property>
       <property name="text">
        <string>  Curve : 0 , Vel : 0</string>
       </property>
       <property name="textFormat">
        <enum>Qt::AutoText</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QPushButton" name="function_3">
       <property name="font">
        <font>
         <pointsize>20</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/images/gui_png/number3.png</normaloff>:/images/gui_png/number3.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>80</width>
         <height>80</height>
        </size>
       </property>
       <property name="flat">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="label3">
       <property name="font">
        <font>
         <family>Noto Sans</family>
         <pointsize>20</pointsize>
         <weight>75</weight>
         <italic>false</italic>
         <bold>true</bold>
         <strikeout>false</strikeout>
         <kerning>true</kerning>
        </font>
       </property>
       <property name="mouseTracking">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">
border-bottom: 1px solid orange;
color : white;
font-family: Noto Sans;
font-style: normal;
line-height: normal;</string>
       </property>
       <property name="text">
        <string>  Curve : 0 , Vel : 0</string>
       </property>
       <property name="textFormat">
        <enum>Qt::AutoText</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QPushButton" name="function_4">
       <property name="font">
        <font>
         <pointsize>20</pointsize>
        </font>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset>
         <normaloff>:/images/gui_png/number4.png</normaloff>:/images/gui_png/number4.png</iconset>
       </property>
       <property name="iconSize">
        <size>
         <width>80</width>
         <height>80</height>
        </size>
       </property>
       <property name="flat">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLabel" name="label4">
       <property name="font">
        <font>
         <family>Noto Sans</family>
         <pointsize>20</pointsize>
         <weight>75</weight>
         <italic>false</italic>
         <bold>true</bold>
         <strikeout>false</strikeout>
         <kerning>true</kerning>
        </font>
       </property>
       <property name="mouseTracking">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">
border-bottom: 1px solid orange;
color : white;
font-family: Noto Sans;
font-style: normal;
line-height: normal;</string>
       </property>
       <property name="text">
        <string>  Curve : 0 , Vel : 0</string>
       </property>
       <property name="textFormat">
        <enum>Qt::AutoText</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="setting">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>10</y>
      <width>60</width>
      <height>60</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>:/images/gui_png/set.png</normaloff>:/images/gui_png/set.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>80</width>
      <height>80</height>
     </size>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>152</x>
      <y>57</y>
      <width>100</width>
      <height>100</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>80</width>
      <height>80</height>
     </size>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>:/images/gui_png/arrow2.png</normaloff>:/images/gui_png/arrow2.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>80</width>
      <height>80</height>
     </size>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_2">
    <property name="geometry">
     <rect>
      <x>62</x>
      <y>147</y>
      <width>100</width>
      <height>100</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>80</width>
      <height>80</height>
     </size>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>:/images/gui_png/arrow4.png</normaloff>:/images/gui_png/arrow4.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>80</width>
      <height>80</height>
     </size>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_3">
    <property name="geometry">
     <rect>
      <x>242</x>
      <y>147</y>
      <width>100</width>
      <height>100</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>80</width>
      <height>80</height>
     </size>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>:/images/gui_png/arrow5.png</normaloff>:/images/gui_png/arrow5.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>80</width>
      <height>80</height>
     </size>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_4">
    <property name="geometry">
     <rect>
      <x>152</x>
      <y>237</y>
      <width>100</width>
      <height>100</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>80</width>
      <height>80</height>
     </size>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>:/images/gui_png/arrow3.png</normaloff>:/images/gui_png/arrow3.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>80</width>
      <height>80</height>
     </size>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_5">
    <property name="geometry">
     <rect>
      <x>152</x>
      <y>147</y>
      <width>100</width>
      <height>100</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>80</width>
      <height>80</height>
     </size>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>:/images/gui_png/home2.png</normaloff>:/images/gui_png/home2.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>100</width>
      <height>100</height>
     </size>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="setting_2">
    <property name="geometry">
     <rect>
      <x>660</x>
      <y>10</y>
      <width>60</width>
      <height>60</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>:/images/gui_png/blue.png</normaloff>:/images/gui_png/blue.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>80</width>
      <height>80</height>
     </size>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="setting_3">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>10</y>
      <width>60</width>
      <height>60</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>:/images/gui_png/wheel.png</normaloff>:/images/gui_png/wheel.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>80</width>
      <height>80</height>
     </size>
    </property>
    <property name="flat">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QLabel" name="label_connect">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>20</y>
      <width>131</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Noto Sans</family>
      <pointsize>15</pointsize>
      <weight>75</weight>
      <italic>false</italic>
      <bold>true</bold>
      <underline>false</underline>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">color : white;
font-family: Noto Sans;
font-style: normal;
line-height: normal;</string>
    </property>
    <property name="text">
     <string>Disconnect</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>function_1</tabstop>
  <tabstop>function_2</tabstop>
  <tabstop>button_left</tabstop>
  <tabstop>function_3</tabstop>
  <tabstop>function_4</tabstop>
  <tabstop>setting</tabstop>
  <tabstop>button_right</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>setting</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>page2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>742</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>686</x>
     <y>43</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>function_1</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>function1()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>461</x>
     <y>102</y>
    </hint>
    <hint type="destinationlabel">
     <x>406</x>
     <y>38</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>function_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>function3()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>418</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>352</x>
     <y>241</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>function_4</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>function4()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>421</x>
     <y>358</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_right</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>right()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>302</x>
     <y>373</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>420</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_left</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>left()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>112</x>
     <y>370</y>
    </hint>
    <hint type="destinationlabel">
     <x>106</x>
     <y>426</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>w()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>193</x>
     <y>106</y>
    </hint>
    <hint type="destinationlabel">
     <x>106</x>
     <y>59</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>d()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>56</x>
     <y>141</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>a()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>308</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>106</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>s()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>201</x>
     <y>267</y>
    </hint>
    <hint type="destinationlabel">
     <x>221</x>
     <y>212</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>home()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>203</x>
     <y>182</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>33</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setting_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>page4()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>679</x>
     <y>55</y>
    </hint>
    <hint type="destinationlabel">
     <x>597</x>
     <y>52</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setting_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>page3()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>624</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel">
     <x>548</x>
     <y>45</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>function_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>function2()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>425</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>348</x>
     <y>159</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>page2()</slot>
  <slot>function1()</slot>
  <slot>function2()</slot>
  <slot>function3()</slot>
  <slot>function4()</slot>
  <slot>left()</slot>
  <slot>right()</slot>
  <slot>w()</slot>
  <slot>a()</slot>
  <slot>s()</slot>
  <slot>d()</slot>
  <slot>home()</slot>
  <slot>page4()</slot>
  <slot>page3()</slot>
 </slots>
</ui>
